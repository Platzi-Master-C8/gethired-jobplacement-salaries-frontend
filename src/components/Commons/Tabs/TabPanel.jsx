import React from 'react';
import PropTypes from 'prop-types';

import Box from '@mui/material/Box';

const TabPanel = (props) => {
    const { children, value, index } = props;
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    value: PropTypes.number.isRequired,
    index: PropTypes.number.isRequired,
};

export default TabPanel;
