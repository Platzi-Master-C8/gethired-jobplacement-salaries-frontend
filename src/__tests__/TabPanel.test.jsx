import React from 'react';
import { render } from '@testing-library/react';

import { TabPanel } from 'Components/Commons/Tabs';

describe('Tests for component <TabPanel />', () => {
    test('Take Snapshot', () => {
        const { asFragment } = render(
            <TabPanel value={0} index={0}>
                Tab Panel
            </TabPanel>,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
