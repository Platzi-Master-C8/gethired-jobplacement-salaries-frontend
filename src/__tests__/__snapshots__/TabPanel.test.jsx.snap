// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for component <TabPanel /> Take Snapshot 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="simple-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      class="MuiBox-root css-1phy807"
    >
      Tab Panel
    </div>
  </div>
</DocumentFragment>
`;
